juriscraper/opinions/united_states/state/colo.py

```diff
@@ -23,25 +26,62 @@ def __init__(self, *args, **kwargs):
         self.url = "https://www.courts.state.co.us/Courts/Supreme_Court/Case_Announcements/"
 
     def _process_html(self) -> None:
-
         date = self.html.xpath(
             ".//p[contains(text(), 'case announcements for')]/text()"
-        )[0]
-        date_str = (
-            re.findall(r"case announcements for (.*) are now", date)[0]
-            .split(",", 1)[1]
-            .strip()
         )
-        for item in self.html.xpath(
-            ".//a[contains(@href, 'Supreme_Court/Opinions/')]"
-        ):
-            docket, name = item.text_content().split(" ", 1)
-            self.cases.append(
-                {
-                    "date": date_str,
-                    "docket": docket,
-                    "name": name,
-                    "url": item.attrib["href"],
-                    "status": "Published",
-                }
+        if date:
+            date_str = (
+                re.findall(r"case announcements for (.*) are now", date[0])[0]
+                .split(",", 1)[1]
+                .strip()
             )
+            for item in self.html.xpath(
+                ".//a[contains(@href, 'Supreme_Court/Opinions/')]"
+            ):
+                link_text = normalize_dashes(item.text_content().strip())
+                cite_match = re.findall(r"(\d{2,4}\s*CO\s*\d+\w?)", link_text)
+                docket_match = re.findall(r"\d+[A-Z]+\d{2,}", link_text)
+                docket = docket_match[0] if docket_match else ""
+                citation = cite_match[0] if cite_match else ""
+                name = (
+                    link_text.replace(docket, "")
+                    .replace(citation, "")
+                    .lstrip(",- ")
+                )
+                self.cases.append(
+                    {
+                        "date": date_str,
+                        "docket": docket,
+                        "name": name,
+                        "url": item.attrib["href"],
+                        "status": "Published",
+                        "citation": citation,
+                    }
+                )
+
+    def extract_from_text(self, scraped_text: str) -> Dict[str, Any]:
+        """Pass scraped text into function and return data as a dictionary
+        Notes for 'Citation':
+            - Reporter key for this court: 'CO'
+            - Type for a state: 2
+        :param scraped_text: Text of scraped content
+        :return: metadata
+        """
+        match_re = r"The Supreme Court of the State of Colorado.*?(?P<citation>\d{4}\s*CO\s*\d+).*?Supreme Court Case No\. (?P<docket>\d+S[A-Z]\d+)"
```

Comment:
I think this pattern is important for the COA because we are grabbing so much content but I think its overly complex for the Supremes.

You could use something like this.  Or you could just split them up.  Which is probably the safer approach.
(?P<citation>\d{4}\s*CO\s*\d+).*?(?P<docket>\d+S[A-Z]\d+)

Reply:
Also- I think its fine to grab the docket here, even though it is on the webpage- if your gut tells you that they aren't reliable on the website.

Reply:
Divided the RE for citation and docket.